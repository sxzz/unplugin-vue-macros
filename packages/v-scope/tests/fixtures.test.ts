import { resolve } from 'node:path'
import { describe } from 'vitest'
import {
  RollupEsbuildPlugin,
  RollupRemoveVueFilePathPlugin,
  RollupVue,
  RollupVueJsx,
  rollupBuild,
  testFixtures,
} from '@vue-macros/test-utils'
import VScope from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    'tests/fixtures/*.vue',
    (args, id) =>
      rollupBuild(id, [
        VScope(),
        RollupVue(),
        RollupVueJsx(),
        RollupRemoveVueFilePathPlugin(),
        RollupEsbuildPlugin({
          target: 'esnext',
        }),
      ]),
    {
      cwd: resolve(__dirname, '..'),
      promise: true,
    }
  )
})
