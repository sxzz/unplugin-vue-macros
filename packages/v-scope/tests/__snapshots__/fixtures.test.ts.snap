// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/basic.vue 1`] = `
"import { ref, createElementVNode, createTextVNode, toDisplayString } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const _sfc_main = {
  __name: 'basic',
  setup(__props) {

const msg = ref('Hello');

return (_ctx, _cache) => {
  

  return (( a= msg.value + \` Vue\` )=> createElementVNode(\\"h1\\", null, [
    createTextVNode(toDisplayString(a), 1 /* TEXT */)
  ]))()
}
}

};
var basic = /*#__PURE__*/_export_sfc(_sfc_main, [__FILE__]);

export { basic as default };
"
`;

exports[`fixtures > tests/fixtures/multi-vscope.vue 1`] = `
"import { ref, openBlock, createElementBlock, Fragment, createElementVNode, createTextVNode, toDisplayString, renderList } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const _sfc_main = {
  __name: 'multi-vscope',
  setup(__props) {

const msg = ref('Hello');

return (_ctx, _cache) => {
  

  return (openBlock(), createElementBlock(Fragment, null, [
    (( a= msg.value + \` Vue\` )=> createElementVNode(\\"h1\\", null, [
      createTextVNode(toDisplayString(a), 1 /* TEXT */)
    ]))(),
    (openBlock(), createElementBlock(Fragment, null, renderList([1, 2, 3], (i) => {
      return (( x= i + 1 )=> createElementVNode(\\"div\\", null, [
        createTextVNode(toDisplayString(x), 1 /* TEXT */)
      ]))()
    }), 64 /* STABLE_FRAGMENT */))
  ], 64 /* STABLE_FRAGMENT */))
}
}

};
var multiVscope = /*#__PURE__*/_export_sfc(_sfc_main, [__FILE__]);

export { multiVscope as default };
"
`;

exports[`fixtures > tests/fixtures/nested.vue 1`] = `
"import { ref, createElementVNode, createTextVNode, toDisplayString } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const _sfc_main = {
  __name: 'nested',
  setup(__props) {

const msg = ref('Hello');

return (_ctx, _cache) => {
  

  return (( a= 1 )=> createElementVNode(\\"div\\", null, [
    (( b= 1 )=> createElementVNode(\\"span\\", null, [
      createTextVNode(toDisplayString(a) + toDisplayString(b), 1 /* TEXT */)
    ]))()
  ]))()
}
}

};
var nested = /*#__PURE__*/_export_sfc(_sfc_main, [__FILE__]);

export { nested as default };
"
`;

exports[`fixtures > tests/fixtures/no-scope.vue 1`] = `
"import { ref, openBlock, createElementBlock, toDisplayString } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const _sfc_main = {
  __name: 'no-scope',
  setup(__props) {

const msg = ref('Hello World!');

return (_ctx, _cache) => {
  return (openBlock(), createElementBlock(\\"h1\\", null, toDisplayString(msg.value), 1 /* TEXT */))
}
}

};
var noScope = /*#__PURE__*/_export_sfc(_sfc_main, [__FILE__]);

export { noScope as default };
"
`;

exports[`fixtures > tests/fixtures/scope-and-else.vue 1`] = `
"import { ref, createElementVNode, createTextVNode, toDisplayString } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const _hoisted_1 = { key: 0 };
const _hoisted_2 = { key: 1 };
const _hoisted_3 = { key: 2 };


const _sfc_main = {
  __name: 'scope-and-else',
  setup(__props) {

const ok = ref(1);

return (_ctx, _cache) => {
  

  return (ok.value == 1)
    ? (( a= 1 )=> createElementVNode(\\"div\\", _hoisted_1, [
        createTextVNode(toDisplayString(a), 1 /* TEXT */)
      ]))()
    : (ok.value == 2)
      ? (( b= 2 )=> createElementVNode(\\"div\\", _hoisted_2, [
          createTextVNode(toDisplayString(b), 1 /* TEXT */)
        ]))()
      : (( c= 3 )=> createElementVNode(\\"div\\", _hoisted_3, [
          createTextVNode(toDisplayString(c), 1 /* TEXT */)
        ]))()
}
}

};
var scopeAndElse = /*#__PURE__*/_export_sfc(_sfc_main, [__FILE__]);

export { scopeAndElse as default };
"
`;

exports[`fixtures > tests/fixtures/scope-and-elseif.vue 1`] = `
"import { ref, createElementVNode, createTextVNode, toDisplayString, createCommentVNode } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const _hoisted_1 = { key: 0 };
const _hoisted_2 = { key: 1 };


const _sfc_main = {
  __name: 'scope-and-elseif',
  setup(__props) {

const ok = ref(1);

return (_ctx, _cache) => {
  

  return (ok.value == 1)
    ? (( a= 1 )=> createElementVNode(\\"div\\", _hoisted_1, [
        createTextVNode(toDisplayString(a), 1 /* TEXT */)
      ]))()
    : (ok.value == 2)
      ? (( b= 2 )=> createElementVNode(\\"div\\", _hoisted_2, [
          createTextVNode(toDisplayString(b), 1 /* TEXT */)
        ]))()
      : createCommentVNode(\\"v-if\\", true)
}
}

};
var scopeAndElseif = /*#__PURE__*/_export_sfc(_sfc_main, [__FILE__]);

export { scopeAndElseif as default };
"
`;

exports[`fixtures > tests/fixtures/scope-and-for.vue 1`] = `
"import { openBlock, createElementBlock, Fragment, createCommentVNode, renderList, createElementVNode, createTextVNode, toDisplayString } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const _sfc_main = {};

function _sfc_render(_ctx, _cache) {
  

  return (openBlock(), createElementBlock(Fragment, null, [
    createCommentVNode(\\" @vue-skip \\"),
    (openBlock(), createElementBlock(Fragment, null, renderList([1, 2, 3], (i) => {
      return (( x= i + 1 )=> createElementVNode(\\"div\\", null, [
        createTextVNode(toDisplayString(x), 1 /* TEXT */)
      ]))()
    }), 64 /* STABLE_FRAGMENT */))
  ], 2112 /* STABLE_FRAGMENT, DEV_ROOT_FRAGMENT */))
}
var scopeAndFor = /*#__PURE__*/_export_sfc(_sfc_main, [['render',_sfc_render],__FILE__]);

export { scopeAndFor as default };
"
`;

exports[`fixtures > tests/fixtures/scope-and-if.vue 1`] = `
"import { ref, createElementVNode, createTextVNode, toDisplayString, createCommentVNode } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const _hoisted_1 = { key: 0 };


const _sfc_main = {
  __name: 'scope-and-if',
  setup(__props) {

const ok = ref(true);

return (_ctx, _cache) => {
  

  return (ok.value)
    ? (( a= 1 )=> createElementVNode(\\"div\\", _hoisted_1, [
        createTextVNode(toDisplayString(a), 1 /* TEXT */)
      ]))()
    : createCommentVNode(\\"v-if\\", true)
}
}

};
var scopeAndIf = /*#__PURE__*/_export_sfc(_sfc_main, [__FILE__]);

export { scopeAndIf as default };
"
`;

exports[`fixtures > tests/fixtures/template-scope.vue 1`] = `
"import { createElementVNode, createCommentVNode, createTextVNode, toDisplayString } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const _sfc_main = {};

function _sfc_render(_ctx, _cache) {
  

  return (( a= 1 )=> createElementVNode(\\"div\\", null, [
    createCommentVNode(\\" @vue-skip \\"),
    createTextVNode(\\" \\" + toDisplayString(a), 1 /* TEXT */)
  ]))()
}
var templateScope = /*#__PURE__*/_export_sfc(_sfc_main, [['render',_sfc_render],__FILE__]);

export { templateScope as default };
"
`;

exports[`fixtures > tests/fixtures/ts.vue 1`] = `
"import { defineComponent, ref, openBlock, createElementBlock, Fragment, createCommentVNode, createElementVNode, createTextVNode, toDisplayString } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"ts\\",
  setup(__props) {
    const msg = ref(\\"Hello\\");
    return (_ctx, _cache) => {
      
      return openBlock(), createElementBlock(
        Fragment,
        null,
        [
          createCommentVNode(\\" @vue-skip \\"),
          (( a= msg.value + \` Vue\` )=> createElementVNode(\\"h1\\", null, [
            createTextVNode(
              toDisplayString(a),
              1
              /* TEXT */
            )
          ]))()
        ],
        2112
        /* STABLE_FRAGMENT, DEV_ROOT_FRAGMENT */
      );
    };
  }
});

var ts = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { ts as default };
"
`;
